# RT-DETR - Métricas del modelo final sobre test 

Se evalúan los modelos RT-DETR (original y denoised) sobre el split de test del dataset de parches, obteniendo las métricas oficiales de precisión, recall, mAP50 y mAP50-95.

## Metricas entrenamiento RT_DTR de parches en imagenes  original en test 1024

In [2]:
from ultralytics import YOLO

model_parches_original_rtdetr = YOLO("/home/salvarado/TFM/resultados/result_RT_DTR/parches_original_1024_50ep_RT_DTR/weights/best.pt")
metrics_parches_original_rtdetr = model_parches_original_rtdetr.val(
    data="/home/salvarado/TFM/Dataset_preparados/dataset_kanji_parches/originales/dataset_parches_originales.yaml",
    
    split="test",
    imgsz=1024,
    device=2
)

print("=== RT-DETR Parches Original - Test ===")
print(f"Precisión: {metrics_parches_original_rtdetr.box.mp:.3f}")
print(f"Recall:    {metrics_parches_original_rtdetr.box.mr:.3f}")
print(f"mAP50:     {metrics_parches_original_rtdetr.box.map50:.3f}")
print(f"mAP50-95:  {metrics_parches_original_rtdetr.box.map:.3f}")

Ultralytics 8.4.61 🚀 Python-3.11.15 torch-2.5.1+cu121 CUDA:2 (NVIDIA GeForce GTX 1080 Ti, 11165MiB)
rt-detr-l summary: 310 layers, 31,985,795 parameters, 0 gradients, 103.4 GFLOPs
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 3241.8±1414.2 MB/s, size: 4568.6 KB)
val: Scanning /home/salvarado/TFM/Dataset_preparados/dataset_kanji_parches/originales/labels/test.cache... 240 images, 34 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 240/240 62.9Mit/s 0.0s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 15/15 1.0s/it 15.7s0.9s
                   all        240       3657      0.831      0.747      0.846      0.526
Speed: 4.0ms preprocess, 50.9ms inference, 0.0ms loss, 0.2ms postprocess per image
Results saved to /home/salvarado/TFM/notebooks_RT_DTR/runs/detect/val-4
=== RT-DETR Parches Original - Test ===
Precisión: 0.831
Recall:    0.747
mAP50:     0.846
mAP50-95:  0.526


## Analisis

Precisión alta (0.831) pero recall bajo (0.747) → el modelo es bastante fiable cuando detecta algo, pero se le escapan muchos kanjis (falsos negativos).
mAP50-95 (0.526) notablemente más bajo que mAP50 (0.846) → cuando detecta correctamente, la caja no siempre queda bien ajustada al kanji real.

## Metricas entrenamiento RT_DTR de parches en imagenes  denoised en test 1024

In [ ]:
from ultralytics import YOLO

model_parches_denoised_rtdetr = YOLO("/home/salvarado/TFM/resultados/result_RT_DTR/parches_denoised_1024_50ep_RT_DTR/weights/best.pt")
metrics_parches_denoised_rtdetr = model_parches_denoised_rtdetr.val(
    data="/home/salvarado/TFM/Dataset_preparados/dataset_kanji_parches/denoised/dataset_parches_denoised.yaml",
    split="test",
    imgsz=1024,
    device=2
)

print("RT-DETR Parches Denoised - Test ")
print(f"Precisión: {metrics_parches_denoised_rtdetr.box.mp:.3f}")
print(f"Recall:    {metrics_parches_denoised_rtdetr.box.mr:.3f}")
print(f"mAP50:     {metrics_parches_denoised_rtdetr.box.map50:.3f}")
print(f"mAP50-95:  {metrics_parches_denoised_rtdetr.box.map:.3f}")

Ultralytics 8.4.61 🚀 Python-3.11.15 torch-2.5.1+cu121 CUDA:2 (NVIDIA GeForce GTX 1080 Ti, 11165MiB)
rt-detr-l summary: 310 layers, 31,985,795 parameters, 0 gradients, 103.4 GFLOPs
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 1328.7±202.5 MB/s, size: 84.8 KB)
val: Scanning /home/salvarado/TFM/Dataset_preparados/dataset_kanji_parches/denoised/labels/test.cache... 240 images, 34 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 240/240 25.8Mit/s 0.0s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 15/15 1.0it/s 15.0s0.8s
                   all        240       3657      0.847      0.883      0.905      0.571
Speed: 3.5ms preprocess, 49.9ms inference, 0.0ms loss, 0.2ms postprocess per image
Results saved to /home/salvarado/TFM/notebooks_RT_DTR/runs/detect/val-5
=== RT-DETR Parches Denoised - Test ===
Precisión: 0.847
Recall:    0.883
mAP50:     0.905
mAP50-95:  0.571


## Analisis

Mejora en todas las métricas respecto a Original, especialmente en recall (0.747→0.883) y mAP50 (0.846→0.905) → el modelo encuentra muchos más kanjis con la versión denoised.
A diferencia de YOLO (donde denoising no cambiaba nada), en RT-DETR sí parece haber una diferencia notable.

## Cálculo de métricas en test - RF-DETR Imagen completa  Original



In [1]:
from ultralytics import YOLO

model_completa_original_rtdetr = YOLO("/home/salvarado/TFM/resultados/result_RT_DTR/imagen_completa_original_1600_50ep_RT_DTR/weights/best.pt")

metrics_completa_original_rtdetr = model_completa_original_rtdetr.val(
    data="/home/salvarado/TFM/Dataset_preparados/dataset_kanji_prep/dataset_original/dataset_original.yaml",
    split="test",
    imgsz=1600,
    device=2
)

print("=== RT-DETR Imagen Completa Original - Test ===")
print(f"Precisión: {metrics_completa_original_rtdetr.box.mp:.3f}")
print(f"Recall:    {metrics_completa_original_rtdetr.box.mr:.3f}")
print(f"mAP50:     {metrics_completa_original_rtdetr.box.map50:.3f}")
print(f"mAP50-95:  {metrics_completa_original_rtdetr.box.map:.3f}")

Ultralytics 8.4.61 🚀 Python-3.11.15 torch-2.5.1+cu121 CUDA:2 (NVIDIA GeForce GTX 1080 Ti, 11165MiB)
rt-detr-l summary: 310 layers, 31,985,795 parameters, 0 gradients, 103.4 GFLOPs
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 1441.7±78.1 MB/s, size: 60054.0 KB)
val: Scanning /home/salvarado/TFM/Dataset_preparados/dataset_kanji_prep/dataset_original/labels/test.cache... 10 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 10/10 687.6Kit/s 0.0s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 26.3s/it 26.3s
                   all         10       2260      0.658      0.364       0.32      0.187
Speed: 8.1ms preprocess, 154.1ms inference, 0.0ms loss, 0.5ms postprocess per image
Results saved to /home/salvarado/TFM/notebooks_RT_DTR/runs/detect/val-6
=== RT-DETR Imagen Completa Original - Test ===
Precisión: 0.658
Recall:    0.364
mAP50:     0.320
mAP50-95:  0.187


## Cálculo de métricas en test - RF-DETR Imagen completa  Denoised



In [2]:
from ultralytics import YOLO

model_completa_denoised_rtdetr = YOLO("/home/salvarado/TFM/resultados/result_RT_DTR/imagen_completa_denoised_1600_50ep_RT_DTR/weights/best.pt")

metrics_completa_denoised_rtdetr = model_completa_denoised_rtdetr.val(
    data="/home/salvarado/TFM/Dataset_preparados/dataset_kanji_prep/dataset_denoised/dataset_denoised.yaml",
    split="test",
    imgsz=1600,
    device=2
)

print("=== RT-DETR Imagen Completa Denoised - Test ===")
print(f"Precisión: {metrics_completa_denoised_rtdetr.box.mp:.3f}")
print(f"Recall:    {metrics_completa_denoised_rtdetr.box.mr:.3f}")
print(f"mAP50:     {metrics_completa_denoised_rtdetr.box.map50:.3f}")
print(f"mAP50-95:  {metrics_completa_denoised_rtdetr.box.map:.3f}")

Ultralytics 8.4.61 🚀 Python-3.11.15 torch-2.5.1+cu121 CUDA:2 (NVIDIA GeForce GTX 1080 Ti, 11165MiB)
rt-detr-l summary: 310 layers, 31,985,795 parameters, 0 gradients, 103.4 GFLOPs
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 2579.9±523.1 MB/s, size: 656.5 KB)
val: Scanning /home/salvarado/TFM/Dataset_preparados/dataset_kanji_prep/dataset_denoised/labels/test.cache... 10 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 10/10 590.7Kit/s 0.0s
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 1/1 9.1s/it 9.1s
                   all         10       2260      0.452      0.172      0.139     0.0562
Speed: 7.0ms preprocess, 151.9ms inference, 0.0ms loss, 0.4ms postprocess per image
Results saved to /home/salvarado/TFM/notebooks_RT_DTR/runs/detect/val-7
=== RT-DETR Imagen Completa Denoised - Test ===
Precisión: 0.452
Recall:    0.172
mAP50:     0.139
mAP50-95:  0.056
